docs/release.md

```diff
@@ -0,0 +1,54 @@
+# Expected workflow for the development to release process
+
+## Branching strategy
+
+- Checkout the `development` branch after cloning (or forking) the repository
+- Create an appropriately named 'feature', 'hotfix', 'patch', etc branch
```

Comment:
It's a little unorthodox to restrict folks' branch names (especially in the ideal case that the pull requests are coming from external forks), but if we want to do this, I think a little more guidance will be necessary.